Use es6 classes

luisvt opened the pull request:
Use es6 classes syntax instead utils.extends

crobinson42 commented:
@luisvt I'm all for this effort to bring the code base up to date with syntax. We need to do something about the test coverage degradation reporting above. Any ideas?

luisvt commented:
The problem was that I added proxied methods definitions directly to the classes. That caused that the new methods were not covered. I changed the tests to cover those new methods
